packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -229,11 +271,15 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
 
   @override
   Widget build(BuildContext context) {
-    final Color effectiveActiveColor = widget.activeColor
-      ?? CupertinoColors.activeBlue;
-    final Color? inactiveColor = widget.inactiveColor;
-    final Color effectiveInactiveColor = inactiveColor
-      ?? CupertinoColors.inactiveGray;
+     // Colors need to be resolved in selected and non selected states separately
+    final Set<WidgetState> activeStates = states..add(WidgetState.selected);
+    final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
```

Comment:
I thought the other comment mentioned that these lines are changed into using `union` and `difference`. But they're now using `..add` and `..remove`. Doesn't the current way modify `states`? For example, if the current state has `WidgetState.selected`, won't `state` become unselected after the build?

Reply:
This is true. I wonder why this has not caused problems in Material `Checkbox` though, because it does seem bad.

Reply:
Yeah, this is what I don't like about [`ToggleableStateMixin`](https://api.flutter.dev/flutter/material/ToggleableStateMixin-mixin.html): it has a [`states`](https://api.flutter.dev/flutter/material/ToggleableStateMixin/states.html) getter that re-constructs a Set each time it's accessed.

So the above code works fine, but it gives the appearance that `activeStates` and `inactiveStates` are modifying the same object.

Reply:
Ah I see. Yeah... the naming is bad here.

Reply:
Maybe add a comment:
```
// The `states` getter constructs a new set every time therefore is safe to edit in place.

Reply:
That'd be great.

Alternatively, we could consider refactoring the getter into a method, so it gives a better intuition:

```diff
- Set<WidgetState> get states {}
+ Set<WidgetState> generateStates() {}
```

Reply:
Yeah, although we should do it in a separate PR since it's potentially breaking.

Reply:
```dart
    final Set<WidgetState> selectedState = <WidgetState>{WidgetState.selected};
    final Set<WidgetState> activeStates = states.union(selectedState);
    final Set<WidgetState> inactiveStates = states.difference(selectedState);
```
How does this looks like, clean and easy to understand.

Reply:
That'd be good, but my guess is that `.union()` and `.difference()` introduce a small performance penalty when compared with `.add()` and `.remove()`.

But if we end up going this route, then `const Set<WidgetState> selectedState` would be ideal.

Reply:
Yeah, I think the current way is ok as long as it's commented.

Reply:
> ```
> // The `states` getter constructs a new set every time, which is therefore safe to edit in place.
> ```

Okay, I have added this comment